modules/effects/arrow-effects/src/main/kotlin/arrow/effects/IORace.kt

```diff
@@ -0,0 +1,114 @@
+package arrow.effects
+
+import arrow.core.*
+import arrow.effects.internal.IOConnection
+import arrow.effects.internal.asyncContinuation
+import arrow.effects.internal.unsafe
+import arrow.effects.typeclasses.Fiber
+import java.util.concurrent.atomic.AtomicBoolean
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.startCoroutine
+import kotlin.coroutines.suspendCoroutine
+
+typealias Pair<A, B> = Either<Tuple2<A, Fiber<ForIO, B>>, Tuple2<Fiber<ForIO, A>, B>>
```

Comment:
FiberPair maybe? Something that doesn't collide.

Reply:
I've seen this somewhere else in the review. Maybe in DeferredK. Maybe we need `FiberPair<F, A, B>`?